tests/Integration/Tagging/TagDocumentTest.php

```diff
@@ -0,0 +1,267 @@
+<?php declare(strict_types=1);
+
+namespace Neusta\Pimcore\HttpCacheBundle\Tests\Integration\Tagging;
+
+use Neusta\Pimcore\HttpCacheBundle\CacheActivator;
+use Neusta\Pimcore\HttpCacheBundle\Element\Document\DocumentType;
+use Neusta\Pimcore\TestingFramework\Database\ResetDatabase;
+use Neusta\Pimcore\TestingFramework\Test\Attribute\ConfigureExtension;
+use Neusta\Pimcore\TestingFramework\Test\Attribute\ConfigureRoute;
+use Neusta\Pimcore\TestingFramework\Test\ConfigurableWebTestcase;
+use Pimcore\Cache\RuntimeCache;
+use Pimcore\Model\Document;
+
+#[ConfigureRoute(__DIR__ . '/Fixtures/get_document_route.php')]
+final class TagDocumentTest extends ConfigurableWebTestcase
+{
+    use ResetDatabase;
+
+    /**
+     * @test
+     */
+    #[ConfigureExtension('neusta_pimcore_http_cache', [
+        'elements' => [
+            'assets' => false,
+            'documents' => true,
+            'objects' => false,
+        ],
+    ])]
+    public function response_is_tagged_with_expected_tags_when_page_is_loaded(): void
+    {
+        $client = self::createClient();
+
+        $document = new Document\Page();
+        $document->setId(42);
+        $document->setKey('test_document_page');
+        $document->setPublished(true);
+        $document->setParentId(1);
+        $document->save();
+
+        // Clear the runtime cache, as it prevents the object from being loaded and thus tagged.
+        // Note: in reality, objects are created and loaded/used in separate requests.
+        RuntimeCache::clear();
+
+        $client->request('GET', '/get-document?id=42');
+
+        $response = $client->getResponse();
+        self::assertSame('Document with key: test_document_page', $response->getContent());
+        self::assertSame(200, $response->getStatusCode());
+        self::assertTrue($response->headers->getCacheControlDirective('public'));
+        self::assertSame('3600', $response->headers->getCacheControlDirective('s-maxage'));
+        self::assertSame('d1,d42', $response->headers->get('X-Cache-Tags'));
```

Comment:
I don't think we should test (published) documents using a custom controller. They are publicly accessible, so we should request the document itself, like this:

```suggestion
        $client->request('GET', '/test_document_page');

        $response = $client->getResponse();
        self::assertSame('Document with key: test_document_page', $response->getContent());
        self::assertSame(200, $response->getStatusCode());
        self::assertTrue($response->headers->getCacheControlDirective('public'));
        self::assertSame('3600', $response->headers->getCacheControlDirective('s-maxage'));
        self::assertSame('d42', $response->headers->get('X-Cache-Tags'));
```